Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -831,6 +832,68 @@ set_option linter.uppercaseLean3 false in
 
 end Semiring
 
+section CommSemiring
```

Comment:
If you don't mind, I would prefer if the changes in this file were postponed till a later PR. It seems to me that these are new additions that are somewhat unrelated to the antiDiagonal refactor.

Reply:
That is possible (but a bit painful, because that postpones once again our main interest for antiDiagonal…)

Reply:
Actually, this is a good and general question:
My impression is that this refactor would be moot without applications like the one for power series. Then, how can one be sure that a PR will lead to such applications if they are not included in it? Chaining pull requests?

Reply:
This refactor is sufficiently justified by the fact it cleans up something in the archive; but indeed, dependent PRs are the preferred way to demonstrate how a more basic result will later be used in something more complicated.

Reply:
Done, see #9309

Reply:
Well, it is possible that I've done it slightly wrong.
- The commit #9309 is a branch, and contains all earlier commits, though after the merge of the present PR, a few modifications will be visible.
- I have also delegated the modifications to the `Archive.Partition` file to that new PR.